pyteal/ast/gitxn.py

```diff
@@ -57,18 +60,35 @@ def __str__(self):
 
     def __teal__(self, options: "CompileOptions"):
         verifyFieldVersion(self.field.arg_name, self.field.min_version, options.version)
-        if type(self.txnIndex) is not int or type(self.index) is not int:
-            raise TealInputError(
-                "Invalid gitxna syntax with immediate transaction index {}, transaction field {}, array index {}".format(
-                    self.txnIndex, self.field, self.index
-                )
-            )
+
+        if type(self.index) is int:
+            opToUse = Op.gitxna
+        else:
+            opToUse = Op.gitxnas
 
         verifyTealVersion(
-            Op.gitxna.min_version, options.version, "TEAL version too low to use gitxna"
+            opToUse.min_version,
+            options.version,
+            "TEAL version too low to use op {}".format(opToUse),
+        )
+
+        if type(self.txnIndex) is int:
+            if type(self.index) is int:
+                op = TealOp(
+                    self, opToUse, self.txnIndex, self.field.arg_name, self.index
+                )
+                return TealBlock.FromOp(options, op)
+            op = TealOp(self, opToUse, self.txnIndex, self.field.arg_name)
+            return TealBlock.FromOp(options, op, cast(Expr, self.index))
+
+        if type(self.index) is int:
+            op = TealOp(self, opToUse, self.field.arg_name, self.index)
+            return TealBlock.FromOp(options, op, cast(Expr, self.txnIndex))
+
+        op = TealOp(self, opToUse, self.field.arg_name)
+        return TealBlock.FromOp(
+            options, op, cast(Expr, self.txnIndex), cast(Expr, self.index)
```

Comment:
nit: if my previous argument about line 50-53 is correct, then you can just use the condition branch between line 76-82, since there's no need in distinguishing `txnIndex`'s type...

Reply:
The line 75 is not necessary, since we no longer need to discuss `self.txnIndex`'s type anymore.

Reply:
@ahangsu Agree - I tried to remove the obsolete logic via https://github.com/algorand/pyteal/pull/193/commits/1e94439f30c8aae8152135cf45cd19a85bbca3a3.  Let me know if it looks good to you.

Reply:
looks correct, this aligns with my thoughts.